packaging/build/Microsoft.NET.Build.Containers.targets

```diff
@@ -94,6 +89,46 @@
+            <ContainerGenerateLabelsImageUrl Condition="'$(ContainerGenerateLabelsImageUrl)' == ''">true</ContainerGenerateLabelsImageUrl>
+            <ContainerGenerateLabelsImageDocumentation Condition="'$(ContainerGenerateLabelsImageDocumentation)' == ''">true</ContainerGenerateLabelsImageDocumentation>
+            <ContainerGenerateLabelsImageSource Condition="'$(ContainerGenerateLabelsImageSource)' == ''">true</ContainerGenerateLabelsImageSource>
+            <ContainerGenerateLabelsImageVersion Condition="'$(ContainerGenerateLabelsImageVersion)' == ''">true</ContainerGenerateLabelsImageVersion>
+            <ContainerGenerateLabelsImageRevision Condition="'$(ContainerGenerateLabelsImageRevision)' == ''">true</ContainerGenerateLabelsImageRevision>
+            <ContainerGenerateLabelsImageVendor Condition="'$(ContainerGenerateLabelsImageVendor)' == ''">true</ContainerGenerateLabelsImageVendor>
+            <ContainerGenerateLabelsImageLicenses Condition="'$(ContainerGenerateLabelsImageLicenses)' == ''">true</ContainerGenerateLabelsImageLicenses>
+            <ContainerGenerateLabelsImageTitle Condition="'$(ContainerGenerateLabelsImageTitle)' == ''">true</ContainerGenerateLabelsImageTitle>
+            <ContainerGenerateLabelsImageBaseDigest Condition="'$(ContainerGenerateLabelsImageBaseDigest)' == ''">true</ContainerGenerateLabelsImageBaseDigest>
+            <ContainerGenerateLabelsImageBaseName Condition="'$(ContainerGenerateLabelsImageBaseName)' == ''">true</ContainerGenerateLabelsImageBaseName>
+        </PropertyGroup>
+
+        <!-- Labels generated from descriptions from the spec at https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys -->
+        <ItemGroup Label="Conventional Label assignment" Condition="'$(ContainerGenerateLabels)' == 'true'">
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageCreated)' == 'true'" Include="org.opencontainers.image.created;org.opencontainers.artifact.created" Value="$([System.DateTime]::UtcNow.ToString('o'))" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageDescription)' == 'true' and '$(Description)' != ''" Include="org.opencontainers.artifact.description;org.opencontainers.image.description" Value="$(Description)" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageAuthors)' == 'true' and '$(Authors)' != ''" Include="org.opencontainers.image.authors" Value="$(Authors)" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageUrl)' == 'true' and '$(PackageProjectUrl)' != ''" Include="org.opencontainers.image.url" Value="$(PackageProjectUrl)" />
+            <ContainerLabel Condition="'$(ContainerGenerateLabelsImageDocumentation)' == 'true' and '$(PackageDocumentationUrl)' != ''" Include="org.opencontainers.image.documentation" Value="$(PackageDocumentationUrl)" />
```

Comment:
Should we use `RepositoryUrl` if `PackageDocumentationUrl` is not present?

>Since $(PackageLicenseUrl) is deprecated, we may just have to default to $(RepositoryUrl). Users should be able to override this easily though.

Reply:
This is a good question, and we probably should do so. `PackageDocumentationUrl` is a thing that is missing from the nuget spec and doesn't exist at all right now. I'm hoping to show it to @JonDouglas and inspire some matching work on the nuspec side - it seems really common in package managers these days to link to documentation for the project, and you can do some cool things in tooling - i.e. IDE F1 keybinding to launch the documentation url, or a `dotnet docs <packageId>` to launch that url in the browser, etc.

Reply:
Yep. Right now we have `projectUrl`, `repository`, and `readme`. Some package managers do have extended metadata for documentation scenarios and better ways to find the `man` of a package. One of the most important parts of a package is its documentation which can be in the package, in the repository of the package, on a project website, etc. Package readmes is one of the major efforts to centralize some of these docs. But today we don't really have a unified way of viewing a package's docset (if they have one).

Reply:
@JonDouglas if/when NuGet _does_ do a thing in this space, would `PackageDocumentationUrl` be a reasonable name? I'm wondering if we in this repo can sort of 'build a foundation' for a future NuGet feature light-up - if we encourage folks to already have this data ready to go, then maybe uptake of the (currently non-existent) NuGet feature would get a boost?

Reply:
Urls have generally been something the team has moved away from. For example `licenseUrl` has been deprecated in favor of `license`. Similar with `iconUrl`. 

I think one could potentially use the `projectUrl` for multiple purposes here given repository's url could include no information at all about the project.